### PR TITLE
Multichain v1 App Footer component

### PR DESCRIPTION
**Pull request by vthomas13**
## Explanation
This PR builds out a new app footer component. Implementation in future story

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/1198

## Manual Testing Steps
For now, mainly use `yarn storybook` and look in the Multichain folder for AppFooter

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/multichain/app-footer/app-footer.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      className="app-footer"
+      width={BlockSize.Full}
+      backgroundColor={BackgroundColor.backgroundDefault}
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      paddingLeft={4}
+      paddingRight={4}
+      paddingTop={2}
+      paddingBottom={2}
+      style={{ bottom: '0px', position: 'fixed', zIndex: 19 }}
```
Comment (darkwing) [inline ui/components/multichain/app-footer/app-footer.js:40]: We can move these styles to stylesheet

File: ui/components/ui/icon-button/icon-button.js [hunk 2]
```diff
@@ -28,25 +28,26 @@ export default function IconButton({
       {renderWrapper(
         <>
           <div className="icon-button__circle">{Icon}</div>
-          {label.length > 9 ? (
-            <Tooltip title={label} position="bottom">
+          {label &&
```
Comment (darkwing) [inline ui/components/ui/icon-button/icon-button.js:31]: This can be `label?.length`

File: ui/pages/routes/routes.component.js [hunk 3]
```diff
@@ -584,6 +585,7 @@ export default class Routes extends Component {
           {isLoading ? <Loading loadingMessage={loadMessage} /> : null}
           {!isLoading && isNetworkLoading ? <LoadingNetwork /> : null}
           {this.renderRoutes()}
+          <AppFooter />
```
Comment (darkwing) [inline ui/pages/routes/routes.component.js:588]: Let's not implement this footer in this PR -- we can do that in a separate PR

File: ui/components/multichain/app-footer/app-footer.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          Icon={
+            <Icon
+              color={IconColor.primaryInverse}
+              name={IconName.SwapVertical}
+              size={IconSize.Lg}
+            />
+          }
+          data-testid="app-Footer-actions-button"
+          onClick={() => {
+            console.log('clicked');
```
Comment (DDDDDanica) [inline ui/components/multichain/app-footer/app-footer.js:87]: Question: is this intended?

File: ui/components/multichain/app-footer/app-footer.stories.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { AppFooter } from '.';
+
+export default {
+  title: 'Components/Multichain/AppFooter',
+};
+// TODO: fix story alignment issues
```
Comment (DDDDDanica) [inline ui/components/multichain/app-footer/app-footer.stories.js:7]: Can we have a dedicated ticket for this TODO?

File: ui/components/multichain/app-footer/app-footer.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        </Text>
+      </Box>
+      <Box
+        width={BlockSize.OneThird}
+        padding={2}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        alignItems={AlignItems.center}
+      >
+        <IconButton
```
Comment (NidhiKJha) [inline ui/components/multichain/app-footer/app-footer.js:76]:
```suggestion
        Let's use ButtonIcon from DS here
```

File: ui/components/multichain/app-footer/app-footer.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      </Box>
+      <Box
+        className="app-footer__button"
+        width={BlockSize.OneThird}
+        padding={2}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        alignItems={AlignItems.center}
+        onClick={() => {
+          history.push(CONNECTED_ROUTE);
```
Comment (NidhiKJha) [inline ui/components/multichain/app-footer/app-footer.js:99]: I don't think we should keep onClick here for connections since it's gonna be a Route

File: ui/components/multichain/app-footer/app-footer.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      >
+        <Icon
+          color={
+            activeConnections
+              ? IconColor.primaryDefault
+              : IconColor.iconAlternative
+          }
+          name={IconName.Global}
+          size={IconSize.Lg}
+        />
```
Comment (NidhiKJha) [inline ui/components/multichain/app-footer/app-footer.js:110]: Can we use ButtonIcon here with href?

File: ui/components/multichain/app-footer/app-footer.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        className="app-footer__button"
+        width={BlockSize.OneThird}
+        padding={2}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        alignItems={AlignItems.center}
+        onClick={() => {
+          history.push(DEFAULT_ROUTE);
+        }}
+      >
```
Comment (NidhiKJha) [inline ui/components/multichain/app-footer/app-footer.js:52]: Can we remove onClick here

File: ui/components/multichain/app-footer/app-footer.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        <Icon
+          color={
+            activeWallet ? IconColor.primaryDefault : IconColor.iconAlternative
+          }
+          name={IconName.Wallet}
+          size={IconSize.Lg}
+        />
```
Comment (NidhiKJha) [inline ui/components/multichain/app-footer/app-footer.js:59]: Re: ButtonIcon

**Review by NidhiKJha — DISMISSED** [review]
Great start! Just need to update some components with the new Design System components

File: ui/components/multichain/app-footer/app-footer.js (hunk 4 above)
Comment (vthomas13) [reply ui/components/multichain/app-footer/app-footer.js:87]: good catch, thanks!

File: ui/components/multichain/app-footer/app-footer.stories.js (hunk 5 above)
Comment (vthomas13) [reply ui/components/multichain/app-footer/app-footer.stories.js:7]: I managed to fix it in my most recent commit, just forgot to take out the comment. Thanks!

**Comment by Ashok1927** [comment]
I like only "TRUE LOGO" OFFICIAL MAIL ONLY...

I don't like any " mail.😢etc...😢😢

I am informing you... I like only "TRUE LOGO" OFFICIAL MAIL ONLY...

Please take carefully...

👌Thanks..

AshokDubey

AshokDubey

**Comment by Ashok1927** [comment]
I like only importen "TRUE LOGO" OFFICIAL MAIL ONLY...

I don't like any " mail.😢 don't vest my imp time.😢😢

I am informing you... I like only "TRUE LOGO" OFFICIAL MAIL ONLY...

Please take carefully.....


👌Thanks..

AshokDubey

AshokDubey

File: ui/components/multichain/app-footer/app-footer.js (hunk 8 above)
Comment (vthomas13) [reply ui/components/multichain/app-footer/app-footer.js:110]: So maybe I'm misunderstanding the suggestion, but it looks like most usages of href are for external links and not for internal routes. I think we should stick to that convention

File: ui/pages/home/home.component.js [hunk 11]
```diff
@@ -24,7 +24,6 @@ import ConnectedAccounts from '../connected-accounts';
 import { Tabs, Tab } from '../../components/ui/tabs';
 import { EthOverview } from '../../components/app/wallet-overview';
 import WhatsNewPopup from '../../components/app/whats-new-popup';
-
```
Comment (darkwing) [inline ui/pages/home/home.component.js:27]: Let's remove this from the PR, it inflates the PR size for no reason.

File: ui/components/ui/icon-button/icon-button.js [hunk 12]
```diff
@@ -57,7 +57,7 @@ IconButton.propTypes = {
   onClick: PropTypes.func.isRequired,
   Icon: PropTypes.object.isRequired,
   disabled: PropTypes.bool,
-  label: PropTypes.string.isRequired,
+  label: PropTypes.string,
```
Comment (darkwing) [inline ui/components/ui/icon-button/icon-button.js:60]: Is this needed?

Comment (vthomas13) [reply ui/components/ui/icon-button/icon-button.js:60]: Actually.. since now we are using button-icon, I think we can get rid of the changes to make label optional. Thanks!

File: ui/components/multichain/app-footer/app-footer.js (hunk 8 above)
Comment (NidhiKJha) [reply ui/components/multichain/app-footer/app-footer.js]: I think for the internal route we don't use target. Otherwise, it's fine to use ButtonLink. But I am fine with an alternative solution

Comment (vthomas13) [reply ui/components/multichain/app-footer/app-footer.js]: fixed in [99edc0c](https://github.com/MetaMask/metamask-extension/pull/20550/commits/99edc0c0fde083a5d19f6660f6afa48c4d31ecc8)

**Review by DDDDDanica — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
